trustfall_core/src/interpreter/execution.rs

```diff
@@ -241,6 +241,24 @@ fn construct_outputs<'query, AdapterT: Adapter<'query>>(
     }))
 }
 
+/// takes a FieldValue and produces a usize clamped to 0 and usize::MAX
+fn usize_from_field_value(field_value: &FieldValue) -> Option<usize> {
+    match field_value {
+        FieldValue::Int64(num) => {
+            Some(usize::try_from(*num.max(&0)).expect("i64 can be converted to usize"))
+        }
+        FieldValue::Uint64(num) => {
+            Some(usize::try_from(*num).expect("i64 can be converted to usize"))
+        }
+        FieldValue::Float64(_)
+        | FieldValue::List(_)
+        | FieldValue::Enum(_)
+        | FieldValue::Boolean(_)
+        | FieldValue::String(_)
+        | FieldValue::Null => None,
```

Comment:
For `null` it's fine to return `None` (imagine a tag being used), but for everything else we should panic instead because that indicates a bug in the type-checker.